src/tools/rustfmt/src/parse/macros/mod.rs

```diff
@@ -15,7 +15,7 @@ pub(crate) mod cfg_if;
 pub(crate) mod lazy_static;
 
 fn build_stream_parser<'a>(sess: &'a ParseSess, tokens: TokenStream) -> Parser<'a> {
-    stream_to_parser(sess, tokens, MACRO_ARGUMENTS)
+    stream_to_parser(sess, tokens, MACRO_ARGUMENTS).recovery(Recovery::Forbidden)
```

Comment:
This will make sure that new recovery code also doesn't affect macro arg parsing. We use this `Recovery::Forbidden` flag when parsing args for macros by example.